packages/@aws-cdk/cdk/lib/core/tokens/cfn-tokens.ts

```diff
@@ -0,0 +1,52 @@
+import { ResolveContext, Token } from "./token";
+
+/**
+ * A Token that represents a CloudFormation reference to another resource
+ */
+export class CfnReference extends Token {
+  /**
+   * The reference type for instances of this class
+   */
+  public static ReferenceType = 'cfn-reference';
```

Comment:
I rather this type marker exposed more idiomatically via something like `CfnReference.isCfnReference(`